soroban-react-dapp/src/components/web3/GreeterContractInteractions.tsx

```diff
@@ -1,138 +1,109 @@
+        setGreeterMessage(result_string) // Store the fetched greeting in state
       } catch (e) {
-        console.error(e)
-        toast.error('Error while fetching greeting. Try again…')
-        setGreeterMessage(undefined)
+        console.error(e) // Log error if fetching fails
+        toast.error('Error while fetching greeting. Try again…') // Show error message on failure
+        setGreeterMessage(undefined) // Clear the greeting message on failure
       } finally {
-        setFetchIsLoading(false)
+        setFetchIsLoading(false) // Set loading state to false after fetching is done
       }
     }
-  },[sorobanContext,contract])
+  }, [sorobanContext, contract])
 
-  useEffect(() => {void fetchGreeting()}, [updateFrontend,fetchGreeting])
+  useEffect(() => { void fetchGreeting() }, [updateFrontend, fetchGreeting]) // Fetch greeting when the component mounts or when updateFrontend changes
 
+  const { activeChain, server, address } = sorobanContext // Destructure properties from the Soroban context
 
-  const { activeChain, server, address } = sorobanContext
-
-  const updateGreeting = async ({ newMessage }: UpdateGreetingValues ) => {
+  const updateGreeting = async ({ newMessage }: UpdateGreetingValues) => {
     if (!address) {
-      console.log("Address is not defined")
-      toast.error('Wallet is not connected. Try again...')
+      toast.error('Wallet is not connected. Try again...') // Show error if wallet is not connected
       return
-    }
-    else if (!server) {
-      console.log("Server is not setup")
-      toast.error('Server is not defined. Unabled to connect to the blockchain')
+    } else if (!server) {
+      toast.error('Server is not defined. Unable to connect to the blockchain') // Show error if server is undefined
       return
-    }
-    else {
-      const currentChain = activeChain?.name?.toLocaleLowerCase()
+    } else {
+      const currentChain = activeChain?.name?.toLocaleLowerCase() // Get the current chain name
       if (!currentChain) {
-        console.log("No active chain")
-        toast.error('Wallet not connected. Try again…')
+        toast.error('Wallet not connected. Try again…') // Show error if no active chain is found
         return
-      }
-      else {
-
-        setUpdateIsLoading(true)
-
+      } else {
+        setUpdateIsLoading(true) // Set loading state to true while updating
         try {
           const result = await contract?.invoke({
-            method: 'set_title',
-            args: [nativeToScVal(newMessage, {type: "string"})],
-            signAndSend: true
+            method: 'set_title', // Invoke the 'set_title' method on the contract
+            args: [
+              nativeToScVal(newMessage, { type: "string" }), // Convert newMessage to a Stellar smart contract value
+              nativeToScVal(address, { type: "address" }) // Convert the user's address to a Stellar smart contract value
+            ],
```

Comment:
If instead of addess I put the harcoded address in the contract, I am able to change the title...

Easy to attack